docs/yarn.md

```diff
@@ -0,0 +1,178 @@
+a new protocol which cachi2 doesn't know about in which case the dependency pre-fetch stage will
+fail with an error._
+
+## Specifying packages to process
+
+A package is a file or directory that is described by a
+[package.json](https://v3.yarnpkg.com/configuration/manifest/) file (also called a
+manifest).
+
+Cachi2 ``fetch-deps`` shell command:
+
+```shell
+cachi2 fetch-deps \
+  --source ./my-repo \
+  --output ./cachi2-output \
+  '<JSON input>'
+```
+
+JSON input:
+```jsonc
+{
+  // "yarn" tells Cachi2 to process Yarn packages
+  "type": "yarn",
+  // path to the package (relative to the --source directory)
+  // defaults to "."
+  "path": ".",
+}
+```
+
+or more simply by just invoking:
+``cachi2 fetch-deps yarn``
+
+For complete example of how to pre-fetch dependencies, see [Pre-fetch dependencies][usage-prefetch].
+
+### Controlling Yarn's behavior
+
+Cachi2 instructs Yarn to download dependencies explicitly declared in ``package.json``. The
+dependencies are then further managed in a ``yarn.lock`` file that Yarn CLI manages automatically
+and creates it if missing. However, **Cachi2 will refuse to process your repository if the file is
+missing**, so be sure to check that file into the repository. Also make sure that the file is up
+to date for which you can use [yarn
+install](https://v3.yarnpkg.com/getting-started/usage/#installing-all-the-dependencies).
+
+### Downloading dependencies
+If Yarn is configured to operate in the [PnP mode](https://v3.yarnpkg.com/features/pnp) (the
+default in Yarn v3) Yarn will store all dependencies as [ZIP
+archives](https://v3.yarnpkg.com/features/pnp/#packages-are-stored-inside-zip-archives-how-can-i-access-their-files).
+
+Once the source repository analysis and verification described in the earlier sections of this
+document has been completed, then it's essentially just a matter of cachi2 internally invoking
+``yarn install --mode=skip-build`` to fetch all dependencies (including transitive dependencies).
+    
+## Using fetched dependencies
+
+See also [usage.md](usage.md) for a complete example of Cachi2 usage.
+
+Cachi2 downloads the Yarn dependencies into the ``deps/yarn/`` subpath of the output directory (see
+the snippet below).
+
+```text
+cachi2-output/deps/yarn
+└── cache
+    ├── abbrev-npm-1.1.1-3659247eab-8.zip
+    ├── agent-base-npm-6.0.2-428f325a93-8.zip
+    ├── agentkeepalive-npm-4.3.0-ac3d8e6807-8.zip
+    ├── aggregate-error-npm-3.1.0-415a406f4e-8.zip
+    ├── ansi-regex-npm-3.0.1-01f44078a3-8.zip
+...
+```
+
+### Known pitfalls
+If your repository isn't in a pristine state (i.e. you tried to run ``yarn install`` previously on
+your own without Cachi2) what may happen is that Cachi2 will assume the repository makes use of
+[Zero-Installs](#dealing-with-yarn-zero-installs). The workaround here is simple, just run ``yarn
+cache clean`` and cachi2 will then process your repository as normal.
```

Comment:
I like this explanation here, but I feel like it belongs in the previous h2 (Specifying packages to process).